crates/bevy_app/src/testing_tools.rs

```diff
@@ -0,0 +1,292 @@
+    /// app.assert_component_eq<()>(Life(9));
+    ///
+    /// app.update();
+    /// // Because all of our entities with the `Life` component also
+    /// // have the `Player` component, these will be equivalent.
+    /// app.assert_component_eq<With<Player>>(Life(10));
+    ///
+    /// app.update();
+    /// // Check that life regeneration caps at 10, as intended
+    /// // Filtering by the component type you're looking for is useless,
+    /// // but it's helpful to demonstrate composing query filters here
+    /// app.assert_component_eq<(With<Player>, With<Life>)>(Life(10));
+    /// ```
+    pub fn assert_component_eq<C, F>(&mut self, value: &C)
+    where
+        C: Component + PartialEq + Debug,
+        F: WorldQuery,
+        <F as WorldQuery>::Fetch: FilterFetch,
+    {
+        self.world.assert_component_eq::<C, F>(value);
+    }
+
+    /// Returns the number of entities found by the [`Query`](bevy_ecs::system::Query) with the type parameters `Q` and `F`
+    ///
+    /// # Example
+    /// ```rust
+    /// # use bevy_app::App;
+    /// # use bevy_ecs::prelude::*;
+    ///
+    /// #[derive(Component)]
+    /// struct Player;
+    ///
+    /// #[derive(Component)]
+    /// struct Life(usize);
+    ///
+    /// let mut app = App::new();
+    ///
+    /// fn spawn_player(mut commands: Commands){
+    ///     commands.spawn().insert(Life(10)).insert(Player);
+    /// }
+    ///
+    /// app.add_startup_system(spawn_player);
+    /// assert_eq!(app.query_len::<&Life, With<Player>>(), 0);
+    ///
+    /// // Run the `Schedule` once, causing our startup system to run
+    /// app.update();
+    /// assert_eq!(app.query_len::<&Life, With<Player>>(), 1);
+    ///
+    /// // Running the schedule again won't cause startup systems to rerun
+    /// app.update();
+    /// assert_eq!(app.query_len::<&Life, With<Player>>(), 1);
+    /// ```
+    pub fn query_len<Q, F>(&mut self)
+    where
+        Q: WorldQuery,
+        F: WorldQuery,
+        <F as WorldQuery>::Fetch: FilterFetch,
+    {
+        self.world.query_len::<Q, F>();
+    }
+
+    /// Sends an `event` of type `E`
+    ///
+    /// # Example
+    /// ```rust
+    /// # use bevy_app::App;
+    /// # use bevy_ecs::prelude::*;
+    ///
+    /// let mut app = App::new();
+    ///
+    /// struct Message(String);
+    ///
+    /// fn print_messages(mut messages: EventReader<Message>){
+    ///     for message in messages.iter(){
+    ///         println!("{}", message.0);
+    ///     }
+    /// }
+    ///
+    /// app.add_event::<Message>().add_system(print_messages);
+    /// app.send_event(Message("Hello!".to_string()));
+    ///
+    /// // Says "Hello!"
+    /// app.update();
+    ///
+    /// // All the events have been processed
+    /// app.update();
+    /// ```
+    pub fn send_event<E: Resource>(&mut self, event: E) {
+        self.world.send_event(event);
+    }
+
+    /// Asserts that the number of events of the type `E` that were sent this frame is exactly `n`
```

Comment:
The description doesn't fit the function.

```suggestion
    /// Returns the number of events of the type `E` that were sent this frame
```